src/ansys/dpf/composites/sampling_point.py

```diff
@@ -323,6 +343,30 @@ def is_uptodate(self) -> bool:
         """True if the results are up-to-date."""
         return self._is_uptodate
 
+    def _get_full_scope(self) -> dpf.Scoping:
+        """Select all solid elements of stack if the selected element is a solid element."""
+        scope = dpf.Scoping()
+        if self._element_info.is_shell:
+            scope.ids = [self.element_id]
+        else:
+            solid_stack_provider = SolidStackProvider(self._meshed_region, self._layup_provider)
+            # the selected element must be the first element in the scope
```

Comment:
Is this always the case, or only here? If it's always needed, could be moved into the `get_solid_stack` code.

Also would be useful to know _why_ it needs to be first.